src/segments/nba.go

```diff
@@ -0,0 +1,257 @@
+	Score             int      `json:"score"`
+	Seed              *int     `json:"seed"`    // Pointer because it can be null
+	InBonus           *string  `json:"inBonus"` // Pointer because it can be null
+	TimeoutsRemaining int      `json:"timeoutsRemaining"`
+	Periods           []Period `json:"periods"`
+}
+
+type Period struct {
+	Period     int    `json:"period"`
+	PeriodType string `json:"periodType"`
+	Score      int    `json:"score"`
+}
+
+type GameLeaders struct {
+	HomeLeaders PlayerLeaders `json:"homeLeaders"`
+	AwayLeaders PlayerLeaders `json:"awayLeaders"`
+}
+
+type PlayerLeaders struct {
+	PersonID    int     `json:"personId"`
+	Name        string  `json:"name"`
+	JerseyNum   string  `json:"jerseyNum"`
+	Position    string  `json:"position"`
+	TeamTricode string  `json:"teamTricode"`
+	PlayerSlug  *string `json:"playerSlug"` // Pointer because it can be null
+	Points      int     `json:"points"`
+	Rebounds    int     `json:"rebounds"`
+	Assists     int     `json:"assists"`
+}
+
+type PbOdds struct {
+	Team      *string `json:"team"` // Pointer because it can be null
+	Odds      float64 `json:"odds"`
+	Suspended int     `json:"suspended"`
+}
+
+// NightscoutData struct contains the API data
+type NBAData struct {
+	HomeTeam       string
+	AwayTeam       string
+	Time           string
+	GameDate       string
+	StartTimeUTC   string
+	GameStatus     int // 1 = scheduled, 2 = in progress, 3 = finished
+	HomeScore      int
+	AwayScore      int
+	HomeTeamWins   int
+	HomeTeamLosses int
+	AwayTeamWins   int
+	AwayTeamLosses int
+}
+
+func (nba *NBA) Template() string {
+	return " 󰠆{{ if eq .GameStatus 1}} {{ .HomeTeam}} ({{.HomeTeamWins}}-{{.HomeTeamLosses}}) vs {{ .AwayTeam}} ({{.AwayTeamWins}}-{{.AwayTeamLosses}}) | {{.GameDate}} | {{.Time}} {{ end }} {{ if gt .GameStatus 1 }} {{ .HomeTeam}} ({{.HomeTeamWins}}-{{.HomeTeamLosses}}):{{.HomeScore}} | {{.AwayTeam}} ({{.AwayTeamWins}}-{{.AwayTeamLosses}}):{{ .AwayScore}} | {{.Time}} {{ end }} "
+}
+
+func (nba *NBA) Enabled() bool {
+	data, err := nba.getResult()
+	if err != nil {
+		return false
+	}
+
+	nba.NBAData = *data
+
+	return true
+}
+
+func (nba *NBA) getResult() (*NBAData, error) {
+	// parses through a set of games and looks for props.team in away or home team
+	findGameByTeamTricode := func(games []Game, teamTricode string) (*Game, error) {
+		for _, game := range games {
+			if game.HomeTeam.TeamTricode == teamTricode || game.AwayTeam.TeamTricode == teamTricode {
+				return &game, nil
+			}
+		}
+		return nil, errors.New("no game found for team")
+	}
+	getCacheValue := func(key string) (*NBAData, error) {
+		val, found := nba.env.Cache().Get(key)
+		// we got something from the cache
+		if found {
+			var nbaData *NBAData
+			err := json.Unmarshal([]byte(val), &nbaData)
+			if err != nil {
+				return nil, err
+			}
+			return nbaData, nil
+		}
+		return nil, errors.New("no data in cache")
+	}
+
+	cachedScheduleKey := nba.props.GetString(TeamName, "") + "schedule"
```

Comment:
team name is fetched twice, I would cache it above and use a `fmt.Sprintf("%sschedule", team)` below for both keys.